notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+       "</table></div>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2023-10-26 01:00:34.553106 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 1, 0, 34, 553106)"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sql='''\n",
+    "SELECT TIME_FLOOR(c.__time, 'P1D') as _date, c.user_id,\n",
+    "  round( TIMESTAMPDIFF(YEAR, TIME_PARSE(u.dob), CURRENT_TIMESTAMP),-1) as age_group, \n",
+    "  count(1) as user_clicks,\n",
+    "  count(1)*100.0/agg.clicks as age_group_contrib_pct\n",
+    "FROM clicks c\n",
+    "     INNER JOIN users u ON c.user_id = u.user_id\n",
+    "     INNER JOIN\n",
+    "     (\n",
+    "         SELECT TIME_FLOOR(cc.__time, 'P1D') as _date,\n",
+    "                ROUND( TIMESTAMPDIFF(YEAR, TIME_PARSE(uu.dob), CURRENT_TIMESTAMP),-1) as age_group,\n",
+    "                COUNT(1) as clicks\n",
+    "           FROM clicks cc\n",
+    "           INNER JOIN users uu ON cc.user_id=uu.user_id\n",
+    "           WHERE cc.__time > CURRENT_TIMESTAMP  - INTERVAL '1' DAY\n",
+    "           GROUP BY 1, ROUND( TIMESTAMPDIFF(YEAR, TIME_PARSE(uu.dob), CURRENT_TIMESTAMP),-1)\n",
+    "     ) agg\n",
+    "        ON agg._date = TIME_FLOOR(c.__time, 'P1D') AND agg.age_group=round( TIMESTAMPDIFF(YEAR, TIME_PARSE(u.dob), CURRENT_TIMESTAMP),-1) \n",
+    "WHERE c.__time > CURRENT_TIMESTAMP  - INTERVAL '1' DAY\n",
+    "GROUP BY 1, 2, round( TIMESTAMPDIFF(YEAR, TIME_PARSE(u.dob), CURRENT_TIMESTAMP),-1), agg.clicks \n",
+    "ORDER BY 1, 5 DESC\n",
+    "LIMIT 10\n",
+    "'''\n",
+    "display.sql(sql)\n",
+    "print_timestamp()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "a96c99be-75c6-410d-9143-0588d887d6ff",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Results = avg:2959.44585 ms   min:2841.148 ms  max:3093.195 ms\n",
+      "2023-10-26 01:01:33.747892 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 1, 1, 33, 747892)"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "result = measure_query(sql, 20)\n",
+    "# saving all results for comparison \n",
+    "join_results.append({\"query_type\":\"subquery_join\",\"results\":result})\n",
+    "print_timestamp()\n",
+    "print(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ee64c9d-4f6d-42b9-827f-4ac01076e00e",
+   "metadata": {},
+   "source": [
+    "### Join using a Lookup\n",
```

Comment:
An example like this is already in the LOOKUPs notebook, I think? So you can remove this and link out to it instead.

Reply:
That would make them open another notebook, find the right section and then make the point about lookup joins being an optimization for queries in some cases.... seems more convoluted that way. Here it follows the sequence of different strategies and contrasts them. So I don't agree.
I understand that you are trying to avoid the same content in multiple places, but I think this is a valid case for that.